proc.c

```diff
@@ -1626,6 +1627,7 @@ mnew_missing(VALUE klass, VALUE obj, ID id, VALUE mclass)
     me = rb_method_entry_create(id, klass, METHOD_VISI_UNDEF, def);
 
     RB_OBJ_WRITE(method, &data->me, me);
+    RB_OBJ_WRITE(method, &data->visibility, METHOD_ENTRY_VISI(me));
```

Comment:
`rb_method_visibility_t` are never `VALUE`s, so you don't want to run write barriers. Plain assignment is enough.  It doesn't happen today, but `rb_method_visibility_t` can change to have numeric values that would make the write barrier think that it's a heap object pointer and do a bad dereference.